baiji/connection.py

```diff
@@ -119,12 +119,12 @@ def cp_r(self, dir_from, dir_to, parallel=False, **kwargs):
                 # Emulate `cp`, which copies the contents of the path.
                 # Get path relative to from_path
                 files_to_copy = [(f, path.join(dir_to, os.path.relpath(path.parse(f).path, from_path)))
-                                 for f in self.ls(dir_from, return_full_urls=True)]
+                                 for f in self.ls(dir_from, return_full_urls=True) if not f.endswith(path.sep)]
```

Comment:
Rather than looking at the string here, it'd be better to use `path.isdirlike`. That way if we ever refine the logic for describing directories then the change gets picked up in places like this without us having to change it in every place.